wip-tmpolaczyk-unhandled-intercept-v2.md

```diff
@@ -0,0 +1,84 @@
+<pre>
+  WIP: WIP-XXXX
+  Layer: Consensus (hard fork)
+  Title: Remove message argument from UnhandledIntercept RADON error
+  Authors: Tomasz Polaczyk <tomasz@witnet.foundation>
+  Discussions-To: `#dev-general` channel on Witnet Community's Discord server
+  Status: Draft
+  Type: Standards Track
+  Created: 2020-09-03
+  License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+The `UnhandledIntercept` RADON error is used to represent RADON script execution errors that do not have a corresponding error code. To help with the identification of unhandled errors, the `UnhandledIntercept` error includes an error message. This message is included in the tally transaction, and it must be agreed upon all the nodes in the network. Therefore, any changes to this message may result in hard forks. This proposal removes the message argument to avoid any future problems.
+
+## Motivation and rationale
+
+During the implementation of [WIP-0017][WIP-0017], we have noticed that renaming a `RadError` may invalidate some tally transactions, creating a hard fork. The reason for that is the `UnhandledIntercept` error which has one extra argument, the "message", which is implemented in Rust as:
+
+```rust
+format!("inner: {:?}", inner)
+```
+
+In other words, it creates a string that contains the debug representation of the inner error. The debug representation is created by the Rust compiler, and it may change between Rust versions. And the inner error is supposed to be an implementation detail of witnet-rust, so it doesn't make much sense for it to be part of consensus-critial code. This gives us a good reason to remove this message.
```

Comment:
```suggestion
In other words, it creates a string that contains the debug representation of the inner error. The debug representation is created by the Rust compiler, and it may change between Rust versions. And the inner error is supposed to be an implementation detail of witnet-rust, so it doesn't make much sense for it to be part of consensus-critical code. This gives us a good reason to remove this message.
```